applications/SwimmingDEMApplication/custom_elements/qsvms_dem_coupled.cpp

```diff
@@ -0,0 +1,859 @@
+/**
+ * @see QSVMSDEMCoupled::EquationIdVector
+ **/
+template < class TElementData >
+void QSVMSDEMCoupled<TElementData>::EquationIdVector(EquationIdVectorType& rResult,
+                                                    ProcessInfo& rCurrentProcessInfo)
+{
+    if (rCurrentProcessInfo[FRACTIONAL_STEP] == 1) {
+        QSVMS<TElementData>::EquationIdVector(rResult, rCurrentProcessInfo);
+        }
+    else {
+        unsigned int LocalIndex = 0;
+        unsigned int lappos = this->GetGeometry()[0].GetDofPosition(VELOCITY_LAPLACIAN_X);
+
+        if (rResult.size() != LocalSize)
+            rResult.resize(LocalSize, false);
+
+        for (unsigned int iNode = 0; iNode < NumNodes; ++iNode)
+            {
+                rResult[LocalIndex++] = this->GetGeometry()[iNode].GetDof(VELOCITY_LAPLACIAN_X,lappos).EquationId();
+                rResult[LocalIndex++] = this->GetGeometry()[iNode].GetDof(VELOCITY_LAPLACIAN_Y,lappos+1).EquationId();
+                if(Dim == 3) rResult[LocalIndex++] = this->GetGeometry()[iNode].GetDof(VELOCITY_LAPLACIAN_Z,lappos+2).EquationId();
+            }
+        }
+}
+
+/**
+ * @see QSVMSDEMCoupled::GetDofList
+ */
+template < class TElementData >
+void QSVMSDEMCoupled<TElementData>::GetDofList(DofsVectorType& rElementalDofList,
+                                                ProcessInfo& rCurrentProcessInfo)
+{
+        if (rCurrentProcessInfo[FRACTIONAL_STEP] == 1) {
+            QSVMS<TElementData>::GetDofList(rElementalDofList, rCurrentProcessInfo);
+
+        }
+        else {
+
+            if (rElementalDofList.size() != LocalSize)
+                rElementalDofList.resize(LocalSize);
+
+            unsigned int LocalIndex = 0;
+
+            for (unsigned int iNode = 0; iNode < NumNodes; ++iNode)
+            {
+                rElementalDofList[LocalIndex++] = this->GetGeometry()[iNode].pGetDof(VELOCITY_LAPLACIAN_X);
+                rElementalDofList[LocalIndex++] = this->GetGeometry()[iNode].pGetDof(VELOCITY_LAPLACIAN_Y);
+                if (Dim == 3) rElementalDofList[LocalIndex++] = this->GetGeometry()[iNode].pGetDof(VELOCITY_LAPLACIAN_Z);
+            }
+        }
+}
+
+template< class TElementData >
+std::string QSVMSDEMCoupled<TElementData>::Info() const
+{
+    std::stringstream buffer;
+    buffer << "QSVMSDEMCoupled #" << this->Id();
+    return buffer.str();
+}
+
+
+template< class TElementData >
+void QSVMSDEMCoupled<TElementData>::PrintInfo(std::ostream& rOStream) const
+{
+    rOStream << "QSVMSDEMCoupled" << Dim << "D";
+}
+
+/**
+ * Returns the squared element size, estimated as h^2 = 2*Area
+ * Returns the squared element size, estimated from the assumption V = (1/6) * h^3
+ * @see QSVMSDEMCoupled::FilterWidth
+ */
+template <class TElementData>
+double QSVMSDEMCoupled<TElementData>::FilterWidth()
+{
+    if (Dim == 2){
+        double FilterWidth = GeometryUtils::CalculateVolume2D(this->GetGeometry());
```

Comment:
Why calling the `GeometryUtils` and not the `DomainSize` in the geometry?  Note that the `GeometryUtils` are only valid for linear triangle and tetrahedra while the QSVMS element should work with quads and hexas.